node_modules/gh-core/lib/server.js

```diff
@@ -61,6 +61,44 @@ var setUpServer = module.exports.setUpServer = function(port, config) {
     // trust the X-Forwarded-* headers
     app.enable('trust proxy');
 
+    // Store the app on which the request was made on the request object
+    app.use(function(req, res, next) {
+        // Pass in a global admin server object when the request was made
+        // on the global admin host
+        if (config.servers.adminHostname === req.hostname) {
+            var globalAdmin = {
+                'displayName': 'Global admin server',
+                'host': req.hostname,
+                'isGlobalAdmin': true
+            };
```

Comment:
Do we have constructors to create these objects instead? Also, what if the AppsAPI took care of doing these admin and shib hostname checks? Then the logic is centralized in a re-usable place.

Something like a single `AppsAPI.getAppByHost(...)` which will do the shib and admin host check, then fall back to the persisted app for the host, if any.